src/odemis/util/img.py

```diff
@@ -1273,3 +1273,19 @@ def apply_flood_fill(input_array, start):
             stack.append((row, col + 1))  # East
 
     return input_array
+
+def max_intensity_projection(data: model.DataArray, axis: int = 0) -> model.DataArray:
+    """Maximum intensity projection of a 3D image along a given axis.
+    :param data: (DataArray) 3D image data (assumed to be ZYX)
+    :param axis: (int) axis along which to project the image. Default is 0 (Z-axis)
+    :return: (DataArray) 2D image data (YX) with the maximum intensity projection
+    """
+    if data.ndim != 3:
+        raise ValueError(f"Data must be 3D, but got {data.ndim} dimensions.")
+    if axis not in (0, 1, 2):
+        raise ValueError(f"Invalid axis {axis}. Must be 0, 1, or 2.")
+
+    proj = numpy.max(data, axis=axis)
```

Comment:
I think we can make this function a bit more flexible, also to facilitate the TODO you added about different projection types:

```suggestion
def intensity_projection(data: model.DataArray, projection_method: Callable[[numpy.ndarray], numpy.ndarray] = numpy.max, axis: int = 0) -> model.DataArray:
    """Intensity projection of a 3D image along a given axis.
    :param data: (DataArray) 3D image data (assumed to be ZYX)
    :param projection_method: (Callable) function to apply to data. Default is numpy.max
    :param axis: (int) axis along which to project the image. Default is 0 (Z-axis)
    :return: (DataArray) 2D image data (YX) with the intensity projection
    """
    if data.ndim != 3:
        raise ValueError(f"Data must be 3D, but got {data.ndim} dimensions.")
    if axis not in (0, 1, 2):
        raise ValueError(f"Invalid axis {axis}. Must be 0, 1, or 2.")

    proj = projection_method(data, axis=axis)
```

Reply:
Please, let's not over-engineer. For now we only need "max".  If one day we see we have many similar functions which only differ on the operator, we can generalize, but not now.
I doubt it's likely to happen, as there are other methods to merge multiple Z layers into a single image, but they don't differ just on the numpy operator. Typically, the other methods are a lot more complex.

Reply:
it's a good suggestion, but we might just shelf it until we actually need other projections. other projection methods aren't really used anywhere as widely as max intensity